[WIP] Fix requeue and to ignore local subscriptions

ShyamsundarR opened the pull request:
- We should requeue the AVR in case there are
multiple subscriptions and one or more of them
fail to create the required manifest work

- We should ignore local subscriptions, typical when
the hub is a managed cluster as well, using the
subscription.status.phase

- Also fixed adding various CRDs to the scheme as
the reconciler failed on a minikube cluster without
the required schemes

- Cleaned up AVR sample to remove placement reference

- Removed StorageClass on PVC sample, to make it easier
to kustomize in a test deployment (as otherwise it keeps
going to the default namespace)

Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

BenamarMk commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -116,19 +115,28 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 	if err := r.updateAVRStatus(ctx, avr, placementDecisions); err != nil {
 		logger.Error(err, "failed to update status")
 
-		return ctrl.Result{}, err
```
I thought returning an error from the reconciler automatically requeues

vdeenadh replied:
That's right: https://github.com/operator-framework/operator-sdk/issues/2898

ShyamsundarR replied:
You are right, I think i changed all returns when looking at this function. We should possibly not return the err and just request for a requeue (the auto stack trace else is a pain in the logs).

BenamarMk commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -105,8 +105,7 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 
-	placementDecisions := r.processSubscriptionList(subscriptionList)
-
+	placementDecisions, requeue := r.processSubscriptionList(subscriptionList)
```
This has been already corrected in the pause/unpause PR.  So you might hold-on to this so that it doesn't cause a merge nightmares.

BenamarMk commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -116,19 +115,28 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+	requeue := false
+
 	for _, subscription := range subscriptionList.Items {
+		if subscription.Status.Phase == subv1.SubscriptionSubscribed {
```
Actually, this was kind of handled in the selectPlacementDecisions by checking for the subv1.SubscriptionPropagated. But this does not hurt having here as well, so I'll going to incorporate this in the my pause/unpuase PR

BenamarMk reviewed (approved):
I approved this PR, so it can go before the pause/unpause PR, which is a big one.  Let's get it in before there.

ShyamsundarR commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -116,19 +115,33 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+		// On the hub ignore any managed cluster subscriptions, as the hub maybe a managed cluster itself.
+		// SubscriptionSubscribed means this subscription is child sitting in managed cluster
+		// Placement.Local is true for a local subscription, and can be used in the absence of Status
+		if subscription.Status.Phase == subv1.SubscriptionSubscribed ||
```
Updated this condition for subscriptions that do not have status updated (happened a couple of times during my testing)

ShyamsundarR commented:
> I approved this PR, so it can go before the pause/unpause PR, which is a big one. Let's get it in before there.

I added one additional check for local placement (and rebased the PR).

@BenamarMk  Should this be merged, or are you taking care of some aspects in your pause/unpause PR?

BenamarMk commented:
@ShyamsundarR it is fine.  I'll merge it and deal with the conflicts later.
